src/hooks/use-welcome-messages.tsx

```diff
@@ -43,11 +43,9 @@ export const WelcomeMessagesProvider = ( { children }: { children: React.ReactNo
 		isFetchingMessages.current = true;
 		try {
 			const response = await client.req.get( {
-				path: '/studio-app/ai-assistant/welcome',
+				path: `/studio-app/ai-assistant/welcome?locale=${ encodeURIComponent( locale ) }`,
 				apiNamespace: 'wpcom/v2',
-				params: { locale },
```

Comment:
The problem with passing the parameter like this is that the request URL gets formatted like so:

  `wpcom/v2/studio-app/ai-assistant/welcome?_locale=pl` with underscore which breaks the request to the API and the translation does not come through

Reply:
Why not adjust the code responsible for that in the custom `request()` function?

https://github.com/Automattic/studio/blob/09f7f3d71288529d9dbf7dedf20bb8ca3fa79975/src/components/auth-provider.tsx#L123

Reply:
Interesting, looking at this setting, I see that when you switch to a language other than English, it appends the locale parameter to all requests going to different endpoints. If I remove the `locale` parameter from the `/welcome` endpoint, I see it also gets appended:

<img width="1260" alt="Captura de pantalla 2024-08-21 a las 9 33 49 a  m" src="https://github.com/user-attachments/assets/1c9a8c29-319d-4927-9aff-1a9380962c4a">

Looking at the original PR where this was introduced - https://github.com/Automattic/local-environment/pull/91 - this seems to have been done so that the locale is set for the duration of the API call. 

In regards to this, I have the following question: 

* Why do we need to set a specific locale for all of these API calls? From my understanding, we are not returning anything locale-specific in any of these endpoints beside the `/welcome` endpoint and even for the `/welcome` endpoint it does not seem to work?

Reply:
> Why do we need to set a specific locale for all of these API calls? From my understanding, we are not returning anything locale-specific in any of these endpoints beside the /welcome endpoint and even for the /welcome endpoint it does not seem to work?

I would think it's to let API endpoints return localized error messages.

Reply:
Update: after Slack discussions, we will proceed with the fix of the API endpoint for the welcome messages. I will update here once it is done

Reply:
@wojtekn I prepared a fix here: D158963-code and it needs to be tested in combination with this PR (the testing steps in the PR are updated). 

Let me know what you think!